### PR TITLE
ptl-tool: print bzs/tickets cited in commit msgs/comments

### PR DESCRIPTION
**Pull request by batrick**
This makes it easier to update issues after a merge.

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

File: src/script/ptl-tool.py [hunk 1]
```diff
@@ -113,6 +113,9 @@
         m = patt.match(line)
         CONTRIBUTORS[m.group(1)] = m.group(2)
 
+BZ_MATCH = re.compile("(.*https?://bugzilla.redhat.com/.*)")
+TRACKER_MATCH = re.compile("(.*https?://tracker.ceph.com/.*)")
```
Comment (joscollin) [inline src/script/ptl-tool.py]: Nit: I think you could safely avoid what ever comes before `http(s)`

See below, the entire commit message is fetched.

```
jcollin@smithi042:~/test-ptl-18547$ src/script/ptl-tool.py --base master --pr-label wip-ptltool1-testing
Adding labeled PR #18284 to PR list
Will merge PRs: [18284]
Detaching HEAD onto base: master
Merging PR #18284
[ 9e4b2868 ] Ceph tracker cited: Testing ptltool commit message.http://tracker.ceph.com/issues/21393.
Checked out new branch wip-jcollin-testing-20171026
Created tag testing/wip-jcollin-testing-20171026_02

```

Comment (batrick) [reply src/script/ptl-tool.py]: We want the whole line that refers to the BZ/tracker so we can see if it says "Fixes: https://bz/" versus "Related-to: https://bz".

**Comment by joscollin** [comment]
@batrick I'm not sure if we use the below cases in Ceph. But I found this:

1. If we have multiple urls in the commit message, it fetches nothing.

For instance,
```
Fixes: http://traker.ceph.com/issues/21393
Fixes: http://traker.ceph.com/issues/21392
```

2. If we have a BZ url, in the commit message, it fetches only the BZ url and not the others.

For instance,
```
Fixes: http://traker.ceph.com/issues/21392
Related-to: https://bugzilla.redhat.com/show_bug.cgi?id=684461
```
fetches only:
```
[ 3c9f2022 ] BZ cited: Related-to: https://bugzilla.redhat.com/show_bug.cgi?id=684461
```

**Comment by batrick** [comment]
> If we have multiple urls in the commit message, it fetches nothing.

You misspelled the URL. It should be `tracker` not `traker`.

Second issue is a consequence of (1).

**Comment by aniyanrajan** [comment]
> <jcollin> Could you please clarify
> <jcollin> What is the need for this change ?
> <jcollin> https://github.com/batrick/ceph/blob/81f8c25bbfaa0e74182669a22b0e697696efd2b6/src/script/ptl-tool.py#L222
> <jcollin> It never printed in my tests !!
> <jcollin> I have never seen a url prefixed: "[ {url} ] Ceph tracker cited
> <jcollin> L226
> <jcollin> Are you fetching URLs from the Reviewer comments, I doubt ?
> <batrick> Yes
> <batrick> It is just going through every comment in the pr
> <jcollin> ok. So what should be the url format in the comments ? "Fixes: <tracker url>" ? I have tried this already, but it never showed up in the output.
> <batrick> Maybe misspelled? :)
> <batrick> Make sure you have the branch checked out too
> <batrick> I sometimes forget myself to do thata
> <jcollin> Let me try again.
> <jcollin> I have created a new PR with a test tracker url in the comment. Commented from my own test github account. http://pastebin.test.redhat.com/528051. The URL is not printed ! Am I missing something here ?
> <jcollin> It fetches the changes successfully, but not fetching the url.
> <batrick> oh, it's not fetching the review itself it hink
> <batrick> ugh
> <batrick> okay fixed
> <batrick> nice find :)

It is working.

**Comment by joscollin** [comment]
> <jcollin> could you please clarify
> <jcollin> What is the need for this change ?
> <jcollin> https://github.com/batrick/ceph/blob/81f8c25bbfaa0e74182669a22b0e697696efd2b6/src/script/ptl-tool.py#L222
> <jcollin> It never printed in my tests !!
> <jcollin> I have never seen a url prefixed: "[ {url} ] Ceph tracker cited
> <jcollin> L226
> <jcollin> Are you fetching URLs from the Reviewer comments, I doubt ?
> <batrick> Yes
> <batrick> It is just going through every comment in the pr
> <jcollin> ok. So what should be the url format in the comments ? "Fixes: <tracker url>" ? I have tried this already, but it never showed up in the output.
> <batrick> Maybe misspelled? :)
> <batrick> Make sure you have the branch checked out too
> <batrick> I sometimes forget myself to do thata
> <jcollin> Let me try again.
> <jcollin> I have created a new PR with a test tracker url in the comment. Commented from my own test github account. http://pastebin.test.redhat.com/528051. The URL is not printed ! Am I missing something here ?
> <jcollin> It fetches the changes successfully, but not fetching the url.
> <batrick> oh, it's not fetching the review itself it hink
> <batrick> ugh
> <batrick> okay fixed
> <batrick> nice find :)

Tested, It is working.
